### DISCUSSION TITLE
Unsupported Message type

### DISCUSSION THREAD

**Discussion by SamSin369** [decentralized-identity/veramo#1063] [Q&A]
![image](https://user-images.githubusercontent.com/69440375/200905729-38cc0788-d1c9-420d-bc5e-8eb50e7d548e.png)
Trying to verify a credential that was created using the handleMessage endpoint of a deployed veramo agent results in this error

**Comment by nickreynolds** [comment]
Sorry for the long delay in responding. This error message can mean many things. You may have already solved this issue by now, but if not I'd be happy to help debug it. If you could copy the `raw` JWT and paste it in to a post, I could take a look at the structure of the credential.

**Comment by mirceanis** [comment]
@SamSin369 Most of the time, this error occurs when the message handler chain is not configured correctly.
Can you post a snippet with your agent configuration? You can include only the message handler part.

It would also help to post a JWT you're trying to verify, as @nickreynolds suggested.

If you're trying to use the `handleMessage` function only to verify a credential, you should be aware that since version 4, veramo supports direct verification of credentials, through the `@veramo/credential-w3c` plugin.
The method is called `verifyCredential`. It's best to recreate the config file after the update, to get the default configuration that includes this new method.
